src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEParameterSymbol.cs

```diff
@@ -702,6 +704,82 @@ private static FlowAnalysisAnnotations DecodeFlowAnalysisAttributes(PEModule mod
             return annotations;
         }
 
+#nullable enable
+        internal override ImmutableArray<int> InterpolatedStringHandlerArgumentIndexes
+        {
+            get
+            {
+                ImmutableArray<int> indexes = _lazyInterpolatedStringHandlerAttributeIndexes;
+                if (indexes.IsDefault)
+                {
+                    indexes = DecodeInterpolatedStringHandlerArgumentAttribute();
+                    Debug.Assert(!indexes.IsDefault);
+                    var initialized = ImmutableInterlocked.InterlockedInitialize(ref _lazyInterpolatedStringHandlerAttributeIndexes, indexes);
+                    Debug.Assert(initialized || indexes.SequenceEqual(_lazyInterpolatedStringHandlerAttributeIndexes));
+                }
+
+                return indexes;
+            }
+        }
+
+        private ImmutableArray<int> DecodeInterpolatedStringHandlerArgumentAttribute()
+        {
+            // If we encounter an invalid attribute, we ignore the contents and just use the empty list.
+
+            if (Type is not NamedTypeSymbol { IsInterpolatedStringHandlerType: true })
+            {
+                return ImmutableArray<int>.Empty;
+            }
+
+            var paramNames = _moduleSymbol.Module.GetInterpolatedStringHandlerArgumentAttributeValues(_handle);
+            Debug.Assert(!paramNames.IsDefault);
+
+            if (paramNames.IsEmpty)
+            {
+                return ImmutableArray<int>.Empty;
+            }
+
+            var builder = ArrayBuilder<int>.GetInstance(paramNames.Length);
+            var parameters = ContainingSymbol switch
+            {
+                MethodSymbol { Parameters: var p } => p,
+                PropertySymbol { Parameters: var p } => p,
+                _ => throw ExceptionUtilities.UnexpectedValue(ContainingSymbol)
+            };
+
+            foreach (var name in paramNames)
+            {
+                switch (name)
+                {
+                    case null:
+                    case { Length: 0 } when ContainingSymbol.IsStatic:
+                        // Invalid data, bail
+                        builder.Free();
+                        return ImmutableArray<int>.Empty;
+
+                    case { Length: 0 }:
+                        builder.Add(-1);
+                        break;
+
+                    default:
+                        var param = parameters.FirstOrDefault(static (p, name) => string.Equals(p.Name, name, StringComparison.Ordinal), name);
```

Comment:
Given that this is the only place we're using FirstOrDefault, is it worth just putting the implementation here, avoiding the lambda, and probably isnt any more lines of code as you don't need the below checks?

Reply:
I don't think so. I prefer it the way I wrote it.